### PR TITLE
Feature/remove event listeners

### PR DESCRIPTION
**Pull request by csolbs24**
# Create Method To Remove Created Event Listeners Without Reloading Page

## Description

Added the `remove_listeners` method so that it is possible to remove all of ULabel's event listeners without reloading the page. Because ULabel instantiates event listeners in multiple ways, this was achieved in multiple ways. First every jquery event listener was added to a .ulabel namespace so that they can all be removed by one jquery .off method. Then an AbortController was added to ulabel and passed as an option into every vanilla js event listener, so that they can be aborted with an abort method. Finally, a reference to every ResizeObserver is stored on ULabel instances so that they can be looped through and terminated.

This is useful for "single page" web applications where ULabel dom elements are removed during navigation but the page is never reloaded. Many ULabel event listeners reference dom elements, but because there is no error checking in place, when ULabel is unable to find them it errors out. This gives an easy way to remove them from outside of ULabel.

## PR Checklist

- [x] Merged latest main
- [x] Version number in `package.json` has been bumped since last release
- [x] Version numbers match between package `package.json` and `src/version.js`
- [x] Ran `npm install` and `npm run build` AFTER bumping the version number
- [x] Updated documentation if necessary (currently just in `api_spec.md`)
- [x] Added changes to `changelog.md`

## Breaking API Changes

None.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -261,7 +285,8 @@ export class ULabel {
         })
 
         // Detection ctrl+scroll
-        document.getElementById(ul.config["annbox_id"]).onwheel = function (wheel_event) {
+        document.getElementById(ul.config["annbox_id"]).addEventListener("wheel", function (wheel_event) {
+            console.log("Wheel event!")
```
Comment (TrevorBurgoyne) [inline src/index.js:289]:
```suggestion
```

File: src/index.js [hunk 2]
```diff
@@ -290,18 +315,33 @@ export class ULabel {
...
+        dialog_resize_observer.observe(document.getElementById(ul.config["imwrap_id"]))
+
+        // Store a reference
+        ul.resize_observers.push(dialog_resize_observer)
+
+        // Create a resize observer to handle toolbox overflow
+        let tb_overflow_resize_observer = new ResizeObserver(function () {
             ul.handle_toolbox_overflow();
-        }).observe(document.getElementById(ul.config["container_id"]));
+            console.log("tb_overflow_resize_observer is observing")
```
Comment (TrevorBurgoyne) [inline src/index.js:334]:
```suggestion
```

**Review by TrevorBurgoyne — COMMENTED** [review]
Very cool, I eagerly await using this 😈

**Review by joshua-dean — APPROVED** [review]
Thank you csolbs24, very cool!

**Review by TrevorBurgoyne — APPROVED** [review]

**Comment by csolbs24** [comment]
I pushed two more commits adding the remaining jQuery event listeners that I missed to the ulabel namespace. Should be good to go now :+1:

**Pull request merged**